packages/react/src/components/SuggestionMenu/DefaultSuggestionMenu.tsx

```diff
@@ -0,0 +1,115 @@
+import { useSuggestionMenuKeyboardNavigation } from "../../hooks/useSuggestionMenuKeyboardNavigation";
+import { useSuggestionMenu } from "../../hooks/useSuggestionMenu";
+import {
+  DefaultRenderItems,
+  SuggestionMenuItemsProps,
+} from "./DefaultRenderItems";
+import { defaultGetItems } from "./defaultGetItems";
+
+export function DefaultPositionedSuggestionMenu<
+  BSchema extends BlockSchema,
+  I extends InlineContentSchema,
+  S extends StyleSchema
+>(props: {
+  editor: BlockNoteEditor<BSchema, I, S>;
+  triggerCharacter?: string;
+  getItems?: (
+    query: string,
+    closeMenu: () => void,
+    clearQuery: () => void
+  ) => Promise<(SuggestionMenuItemProps | SuggestionMenuLabelProps)[]>;
+  renderItems?: FC<SuggestionMenuItemsProps>;
+}) {
+  const { editor, triggerCharacter, getItems, renderItems } = props;
+
+  const { isMounted, suggestionMenuProps, positionerProps } = useSuggestionMenu(
+    editor,
+    triggerCharacter || "/"
+  );
+
+  if (!isMounted) {
+    return null;
+  }
+
+  return (
+    <div ref={positionerProps.ref} style={positionerProps.styles}>
+      <DefaultSuggestionMenu
+        editor={editor}
+        {...suggestionMenuProps}
+        getItems={getItems}
+        renderItems={renderItems}
+      />
+    </div>
+  );
+}
+
+// TODO: The reason these 2 components are split is because the hooks in
+//  `DefaultSuggestionMenu` assume that the menu is open. Therefore, they should
+//  only be run when the menu is open (you cannot conditionally run hooks).
+//  We could add a `show` param to each hook, but I feel like that is less
+//  "React-ish" than splitting the components. I think that it might be an issue
+//  that the menu plugins currently send both position and state data in the
+//  same update, as we can't separate `useSuggestionMenu` into 2 hooks, one for
+//  state and one for position.
+export function DefaultSuggestionMenu<
+  BSchema extends BlockSchema,
+  I extends InlineContentSchema,
+  S extends StyleSchema
+>(props: {
+  editor: BlockNoteEditor<BSchema, I, S>;
+  query: string;
+  closeMenu: () => void;
+  clearQuery: () => void;
+  getItems?: (
+    query: string,
+    closeMenu: () => void,
+    clearQuery: () => void
+  ) => Promise<(SuggestionMenuItemProps | SuggestionMenuLabelProps)[]>;
+  renderItems?: FC<SuggestionMenuItemsProps>;
+}) {
+  const { editor, query, closeMenu, clearQuery, getItems, renderItems } = props;
+
+  const getItemsForLoading = useMemo(
+    () => (query: string) =>
+      getItems !== undefined
+        ? getItems(query, closeMenu, clearQuery)
+        : defaultGetItems(editor, query, closeMenu, clearQuery),
+    [clearQuery, closeMenu, editor, getItems]
+  );
+
+  const { items, usedQuery, loadingState } = useLoadSuggestionMenuItems(
+    query,
+    getItemsForLoading
+  );
+
+  useCloseSuggestionMenuNoItems(items, usedQuery, closeMenu);
+
+  const selectedIndex = useSuggestionMenuKeyboardNavigation(
+    editor,
+    items.filter((item) => !("label" in item)),
```

Comment:
I think you'll run into this when splitting the type of `DefaultPositionedSuggestionMenu`, but I don't think the labels should be returned from `getItems`. Maybe the rendering of the labels should be part of the UI layer?

Reply:
That's one thing I forgot to discuss in the call - there are trade offs with both options:

Including labels in `getItems` allows us to use the same type for items and component props, but means we have to introduce more logic for keyboard navigation.

Not including them means we have to add smth like the old `group` attribute to the default return type of `getItems`, but it does simplify the keyboard navigation since we don't have to account for the labels. It also means we have to sort the items before returning them.

Right now, we also include `icon` in the default item type, so the separation of data/UI is already iffy, so imo it's better to include labels in `getItems`.